src/org/omegat/gui/dictionaries/DictionariesTextArea.java

```diff
@@ -242,42 +256,50 @@ protected void setFoundResult(final SourceTextEntry se, final List<DictionaryEnt
             scrollPane.notify(true);
         }
 
+        final int old = scrollPane.getVerticalScrollBarPolicy();
+        scrollPane.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_ALWAYS);
         StringBuilder txt = new StringBuilder();
         boolean wasPrev = false;
         int i = 0;
+        txt.append("<html>");
         for (DictionaryEntry de : data) {
             if (wasPrev) {
                 txt.append("<br><hr>");
             } else {
                 wasPrev = true;
             }
-            txt.append("<b><span id=\"" + i + "\">");
+            txt.append("<div id =\"" + i + "\"><b><span class=\"word\">");
             txt.append(de.getWord());
             txt.append("</span></b>");
             txt.append(" - ").append(de.getArticle());
-
+            txt.append("</div>");
             displayedWords.add(de.getWord().toLowerCase());
             i++;
-            if (i % TEXT_BATCH == 0) {
-                appendText(txt.toString());
-                txt = new StringBuilder();
-            }
         }
+        txt.append("</html>");
         appendText(txt.toString());
+        scrollPane.setVerticalScrollBarPolicy(old);
     }
 
     private void appendText(final String txt) {
         Document doc = getDocument();
-        if (doc.getLength() == 0) {
+        EditorKit kit = getEditorKit();
+        try {
+            Reader r;
+            if (doc.getLength() == 0) {
+                r = new StringReader(txt);
+            } else {
+                StringBuilder sb = new StringBuilder(doc.getText(0, doc.getLength())).append(txt);
+                r = new StringReader(sb.toString());
+            }
             // Appending to an empty document results in treating HTML tags as
             // plain text for some reason
```

Comment:
I think this comment is not needed anymore?

Reply:
Yes, we can drop it.